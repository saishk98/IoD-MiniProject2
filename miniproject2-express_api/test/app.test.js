const request = require("supertest");
const express = require("express");

const app = express();

app.get("/", (_, res) => {
  res.status(200).send("Express API is running!");
});

describe("GET /", () => {
  it("should return a success message and match snapshot", async () => {
    const response = await request(app).get("/");
    expect(response.status).toBe(200);
    expect(response.text).toBe("Express API is running!");
    expect(response).toMatchSnapshot(); // ðŸ”¹ Adds snapshot testing
  });
});
