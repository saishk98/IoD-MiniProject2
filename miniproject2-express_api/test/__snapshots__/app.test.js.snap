// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET / should return a success message and match snapshot 1`] = `
{
  "header": {
    "connection": "close",
    "content-length": "23",
    "content-type": "text/html; charset=utf-8",
    "date": "Sat, 17 May 2025 08:17:40 GMT",
    "etag": "W/"17-aLgzgx4649ilGIhxMiThDscxwfc"",
    "x-powered-by": "Express",
  },
  "req": {
    "data": undefined,
    "headers": {},
    "method": "GET",
    "url": "http://127.0.0.1:64671/",
  },
  "status": 200,
  "text": "Express API is running!",
}
`;
